lib/rucio/core/rule.py

```diff
@@ -1285,11 +1289,24 @@ def update_rule(rule_id, options, session=None):
 
             elif key == 'child_rule_id':
                 # Check if the child rule has the same scope/name as the parent rule
-                child_rule = session.query(models.ReplicationRule).filter_by(id=options[key]).one()
-                if rule.scope != child_rule.scope or rule.name != child_rule.name:
-                    raise InputValidationError('Parent and child rule must be set on the same dataset.')
-                if child_rule.state != RuleState.OK:
-                    rule.child_rule_id = options[key]
+                child_id: Union[str, None] = options[key]
+                if child_id is None:
+                    # dissolve exisiting relationship
+                    if not rule.child_rule_id:
+                        raise InputValidationError('Cannot detach child when '
+                                                   'no such relationship exists')
```

Comment:
I’m starting to feel strongly against breaking string with errors message for the sake of maintaining a maximum line length. It hinders searching for them with `grep`. I’m not convinced that it is worthwhile.